engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala

```diff
@@ -0,0 +1,200 @@
+package cromwell.engine.workflow.workflowstore
+
+import java.time.OffsetDateTime
+
+import akka.actor.{ActorLogging, LoggingFSM, Props}
+import cromwell.core.{WorkflowId, WorkflowSourceFiles}
+import cromwell.database.obj.WorkflowMetadataKeys
+import cromwell.services.MetadataServiceActor.{MetadataPutAcknowledgement, PutMetadataAction}
+import cromwell.services.{MetadataEvent, MetadataKey, MetadataValue, ServiceRegistryClient}
+import cromwell.engine.workflow.workflowstore.WorkflowStoreActor._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+import scalaz.NonEmptyList
+
+class InMemoryWorkflowStoreDatabase extends WorkflowStoreDatabase {
+
+  var workflowStore = List.empty[SubmittedWorkflow]
+
+  /**
+    * Adds the requested WorkflowSourceFiles to the store and returns a WorkflowId for each one (in order)
+    * for tracking purposes.
+    */
+  override def add(sources: NonEmptyList[WorkflowSourceFiles])(implicit ec: ExecutionContext): Future[NonEmptyList[WorkflowId]] = {
+    val submittedWorkflows = sources map { SubmittedWorkflow(WorkflowId.randomId(), _, Submitted) }
+    workflowStore = workflowStore ++ submittedWorkflows.list
+    Future.successful(submittedWorkflows map { _.id })
+  }
+
+  /**
+    * Retrieves up to n workflows which have not already been pulled into the engine and sets their pickedUp
+    * flag to true
+    */
+  override def fetchRunnableWorkflows(n: Int)(implicit ec: ExecutionContext): Future[List[WorkflowToStart]] = {
+    val startableWorkflows = workflowStore filter { _.state.isStartable } take n
```

Comment:
It should be giving precedence to older submissions (ie FIFO behavior), which on startup *should* be `restartable` > `submitted`